ext/java/nokogiri/internals/NokogiriHelpers.java

```diff
@@ -730,7 +741,7 @@ public class NokogiriHelpers
     if (ruby_encoding == null) { return str; }
     Charset encoding = Charset.forName(ruby_encoding);
     if (Charset.forName(parsed_encoding).compareTo(encoding) == 0) { return str; }
-    if (str.length() == 0) { return str; } // no need to convert
+    if (str.isEmpty()) { return str; } // no need to convert
```

Comment:
For whatever reason, this is failing in the rake-compiler-dock:1.9.1-jruby container with:

```
ext/java/nokogiri/internals/NokogiriHelpers.java:744: error: cannot find symbol
    if (str.isEmpty()) { return str; } // no need to convert
           ^
  symbol:   method isEmpty()
  location: variable str of type CharSequence
```

See https://github.com/sparklemotion/nokogiri/actions/runs/13933282491/job/38995513854?pr=3478#step:4:137 for the CI failure.

I'm afraid I don't understand why this is, and I can't reproduce it on my normal development environment even if I use the same version of JRuby. But I'm going to add a commit to revert this, then this PR should be green.

Reply:
Oops, yep that was a method added in Java 15. The revert is correct.